### PR TITLE
fix: fixing macos ci by redacting the vm_kind

### PR DESCRIPTION
**Pull request by wacban**
the insta snapshots tests work by generating snapshot files containing some configuration
sadly that configuration contains the architecture depended field vm_kind - it's different on macos and linux
this PR disables those tests selectively on macos in order to fix the ci

### CODE REVIEW

**Comment by wacban** [comment]
you can see an example of a failing build here:
https://github.com/near/nearcore/actions/runs/6707949701/job/18227634268

**Comment by wacban** [comment]
@nagisa Once it's merged and hopefully fixes the ci, can we make the macos step required?

**Comment by frol** [comment]
@wacban Have you considered using "redactions" feature of insta instead of banning the test completely? https://insta.rs/docs/redactions/

**Comment by wacban** [comment]
>@wacban Have you considered using "redactions" feature of insta instead of banning the test? https://insta.rs/docs/redactions/ 

Thanks that is much better!

File: core/primitives/src/runtime/snapshots/near_primitives__runtime__config_store__tests__0.json.snap [hunk 1]
```diff
@@ -172,7 +172,7 @@ expression: config_view
     },
     "grow_mem_cost": 1,
     "regular_op_cost": 3856371,
-    "vm_kind": "Wasmer0",
+    "vm_kind": "vm_kind",
```
Comment (nagisa) [inline core/primitives/src/runtime/snapshots/near_primitives__runtime__config_store__tests__0.json.snap:175]: You will need to run the relevant test repeatedly to update the snapshots for all the protocol versions.

File: core/primitives/src/views.rs [hunk 2]
```diff
@@ -2935,7 +2935,7 @@ mod tests {
         let config_store = RuntimeConfigStore::new(None);
         let config = config_store.get_config(PROTOCOL_VERSION);
         let view = RuntimeConfigView::from(RuntimeConfig::clone(config));
-        insta::assert_json_snapshot!(&view);
+        insta::assert_json_snapshot!(&view, { ".wasm_config.vm_kind" => "vm_kind"});
```
Comment (nagisa) [inline core/primitives/src/views.rs:2938]: I would use something like this instead to make it clear the value has been tampered with:


```suggestion
        insta::assert_json_snapshot!(&view, { ".wasm_config.vm_kind" => "<REDACTED>"});
```

**Review by nagisa — APPROVED** [review]

**Pull request merged**